Sources/CombineBluetooth/Internal/CoreBluetoothCentralManager.swift

```diff
@@ -154,18 +157,29 @@ extension CoreBluetoothCentralManager: CentralManager {
     func retrieveConnectedPeripherals(withServices serviceUUIDs: [CBUUID]) -> [BluetoothPeripheral] {
         centralManager.retrieveConnectedPeripherals(withServices: serviceUUIDs).map(peripheral(for:))
     }
+    
+    func registerProxyDelegate(_ proxyDelegate: any CBCentralManagerDelegate) {
+        self.proxyDelegate = proxyDelegate
+    }
+    
+    func unregisterProxyDelegate() {
+        proxyDelegate = nil
+    }
 }
 
 extension CoreBluetoothCentralManager: CBCentralManagerDelegate {
     func centralManagerDidUpdateState(_ central: CBCentralManager) {
+        proxyDelegate?.centralManagerDidUpdateState(central)
```

Comment:
(Obacht, dogmatic oppinion) Can't we implement a publisher that is fed from the underscore subjects and emits these events? Would be not that straight forward like the delegate implementation but this is CombineBluetooth, so everything should be a publisher.

Reply:
Far point, but for the peripheral we did the same and we need to exact calls for the proxy delegation to the SDK we're working with. Otherwise we'd need to build around storing the peripheral anywhere to feed it correctly.

Reply:
Thanks @luizmb  for your input. I introduces such a protocol and refactored a few bits and pieces. Every feedback is welcome.